codility_training/lessons.lesson02.Arrays.CyclicRotation/khw970421.js

```diff
@@ -0,0 +1,9 @@
+function solution(A, K) {
+    // write your code in JavaScript (Node.js 8.9.4)4
+    let b=[];
```

Comment:
[] 객체를 담는 배열 b는 값이 아닌 주소값을 저장하고 있습니다.
b = [1,2] 처럼 직접적으로 수정하지 않고 b[1] = 1; 처럼 수정할 때에는 주소값에 영향을 주지 않습니다 
따라서 이 상황에서는 const 를 사용합니다
```suggestion
    const b=[];
```

Reply:
객체를 담는 배열 b가 주소를 저장한다 하는데
``` javacscript
		const a=[];
		a[1]=10;
		a[2]=100;
		console.log(a[1]);
		a[1]=1;
		console.log(a[1]);
```
에서 값을 넣으면 주소값이 아닌 값이 출력되지않습니까?

Reply:
배열 객체를 담는 A는 주소를 담고 있지만 배열은 요소로 1이라는 값을 담고 있어요
@JunilHwang

Reply:
let으로 선언해야 될 경우는 다음과 같습니다.
```
let a = []
a = 10
```
이렇게 `a`라는 변수에 직접 값을 할당할 때 사용하는 것이고,

```
const a = []
a.push(1)
a[1] = 10
```
이럴 경우 a의 값을 직접 변경하는 것이 아니라, **a가 가르키고 있는 배열의 값**을 변경하는 것이기 때문에 `const`로 사용합니다.